### PR TITLE
Extraneous `waitOnline` calls with `InboundAgentRule`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins-test-harness/pull/577#discussion_r1198023319, minor cleanup after #8002.

### Testing done

Looked for usages of `InboundAgentRule` which also called `waitOnline` in the same JVM session, and deleted that call. Ran the modified tests.

### Proposed changelog entries

- N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8021"></a>

### CODE REVIEW

File: test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java
Comment (jglick) [inline test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java:103]: from https://github.com/jenkinsci/jenkins/blob/2acd833a7cb745d13af6a0fd78661eb771de83d0/test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java#L76-L81

File: test/src/test/java/jenkins/slaves/restarter/JnlpSlaveRestarterInstallerTest.java
Comment (jglick) [inline test/src/test/java/jenkins/slaves/restarter/JnlpSlaveRestarterInstallerTest.java:72]: https://github.com/jenkinsci/jenkins/blob/2acd833a7cb745d13af6a0fd78661eb771de83d0/test/src/test/java/jenkins/slaves/restarter/JnlpSlaveRestarterInstallerTest.java#L78-L81 still needed

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
